src/views/Dashboard/Dashboard.tsx

```diff
@@ -9,13 +29,472 @@ import TopCards from "./components/TopCards";
+    const change24Sushi = numeral(sushiValues.market_data.price_change_percentage_24h_in_currency.usd).format("0.00a") + "%";
+
+    const reservesHistory = [];
+
+    for (let key in assetsHistory.WETH) {
+      const historyObject = {
+        info: "",
+        block: 0,
+        yUSD: 0,
+        DPI: 0,
+        WETH: 0,
+        INDEXLP: 0,
+        INDEX: 0,
+        Sushi: 0,
+        UMA: 0
+      };
+      if (hasKey(assetsHistory.WETH, key)) {
+        const wethValue:any = assetsHistory.WETH[key];
+        if (key !== 'latest') {
+          historyObject.info = "Block " + key;
+          historyObject.block = key;
+          historyObject.WETH = wethValue.value;
+          if (hasKey(assetsHistory.yUSD, key)) {
+            const yUSDValue:any = assetsHistory.yUSD[key];
+            historyObject.yUSD = yUSDValue.value;
+          }
+          if (hasKey(assetsHistory.DPI, key)) {
+            const dpiValue:any = assetsHistory.DPI[key];
+            historyObject.DPI = dpiValue.value;
+          }
+          if (hasKey(assetsHistory.INDEX, key)) {
+            const indexValue:any = assetsHistory.INDEX[key];
+            historyObject.INDEX = indexValue.value;
+          }
+          if (hasKey(assetsHistory.INDEXLP, key)) {
+            const indexLPValue:any = assetsHistory.INDEXLP[key];
+            historyObject.INDEXLP = indexLPValue.value;
+          }
+          if (hasKey(assetsHistory.Sushi, key)) {
+            const sushiValue:any = assetsHistory.Sushi[key];
+            historyObject.Sushi = sushiValue.value;
+          }
+          if (hasKey(assetsHistory.UMA, key)) {
+            const umaValue:any = assetsHistory.UMA[key];
+            historyObject.UMA = umaValue.value;
+          }
+        } else {
+          historyObject.info = "present";
+          historyObject.block = currentBlock + 5000;
+          historyObject.WETH = wethValue;
+          if (hasKey(assetsHistory.yUSD, key)) {
+            const yUSDValue:any = assetsHistory.yUSD[key];
+            historyObject.yUSD = yUSDValue;
+          }
+          if (hasKey(assetsHistory.DPI, key)) {
+            const dpiValue:any = assetsHistory.DPI[key];
+            historyObject.DPI = dpiValue;
+          }
+          if (hasKey(assetsHistory.INDEX, key)) {
+            const indexValue:any = assetsHistory.INDEX[key];
+            historyObject.INDEX = indexValue;
+          }
+          if (hasKey(assetsHistory.INDEXLP, key)) {
+            const indexLPValue:any = assetsHistory.INDEXLP[key];
+            historyObject.INDEXLP = indexLPValue;
+          }
+          if (hasKey(assetsHistory.Sushi, key)) {
+            const sushiValue:any = assetsHistory.Sushi[key];
+            historyObject.Sushi = sushiValue;
+          }
+          if (hasKey(assetsHistory.UMA, key)) {
+            const umaValue:any = assetsHistory.UMA[key];
+            historyObject.UMA = umaValue;
+          }
+        }
+      }
+      reservesHistory.push(historyObject);
+    }
```

Comment:
each time we need to add a new asset, we need to edit this adding a new code into it?
- doing this in a dynamic way would be rather good